frontend/src/modules/organization/pages/organization-list-page.vue

```diff
@@ -86,109 +86,198 @@
-import { useLfSegmentsStore } from '@/modules/lf/segments/store';
-import LfIcon from '@/ui-kit/icon/Icon.vue';
-import LfButton from '@/ui-kit/button/Button.vue';
+import { useOrganizationStore } from '@/modules/organization/store/pinia';
+import AppPageWrapper from '@/shared/layout/page-wrapper.vue';
+import LfFilter from '@/shared/modules/filters/components/Filter.vue';
+import { filterApiService } from '@/shared/modules/filters/services/filter-api.service';
+import { FilterQuery } from '@/shared/modules/filters/types/FilterQuery';
 import usePermissions from '@/shared/modules/permissions/helpers/usePermissions';
 import { LfPermission } from '@/shared/modules/permissions/types/Permissions';
-import LfOrganizationAdd from '@/modules/organization/components/edit/organization-add.vue';
-import allOrganizations from '@/modules/organization/config/saved-views/views/all-organizations';
+import LfSavedViews from '@/shared/modules/saved-views/components/SavedViews.vue';
+import { TanstackKey } from '@/shared/types/tanstack';
+import LfButton from '@/ui-kit/button/Button.vue';
+import LfIcon from '@/ui-kit/icon/Icon.vue';
+import { useQuery, useQueryClient } from '@tanstack/vue-query';
+import { storeToRefs } from 'pinia';
+import {
+  computed, onMounted, ref, watch,
+} from 'vue';
+
+const { buildApiFilter } = filterApiService();
 
 const organizationStore = useOrganizationStore();
-const { filters, totalOrganizations, savedFilterBody } = storeToRefs(organizationStore);
-const { fetchOrganizations } = organizationStore;
+const { filters } = storeToRefs(organizationStore);
 
-const loading = ref(true);
-const tableLoading = ref(true);
-const isSubProjectSelectionOpen = ref(false);
 const organizationCreate = ref(false);
 
-const organizationFilter = ref<LfFilter | null>(null);
-const lsSegmentsStore = useLfSegmentsStore();
+const organizationFilter = ref<InstanceType<typeof LfFilter> | null>(null);
 
+const lsSegmentsStore = useLfSegmentsStore();
 const { selectedProjectGroup } = storeToRefs(lsSegmentsStore);
 
 const { hasPermission } = usePermissions();
 
+const queryClient = useQueryClient();
+
 const pagination = ref({
   page: 1,
   perPage: 20,
 });
 
-filters.value = { ...allOrganizations.config };
+// Reactive state for query parameters
+const queryParams = ref({
+  search: '',
+  filter: filters.value,
+  offset: 0,
+  limit: 20,
+  orderBy: 'activityCount_DESC',
+  segments: selectedProjectGroup.value?.id ? [selectedProjectGroup.value.id] : [],
+});
 
-const showLoading = (filter: any, body: any): boolean => {
-  const saved: any = { ...savedFilterBody.value };
-  delete saved.offset;
-  delete saved.limit;
-  delete saved.orderBy;
-  const compare = {
-    ...body,
-    filter,
-  };
-  return JSON.stringify(saved) !== JSON.stringify(compare);
-};
+// Create a computed query key for organizations
+const organizationsQueryKey = computed(() => [
+  TanstackKey.ORGANIZATIONS_LIST,
+  selectedProjectGroup.value?.id,
+  {
```

Comment:
Same as comment above im afraid this wont trigger update in case any variable changes because you are using nested object
{
key: value
anotherKey: value2
}

to 
[
value,
value2
]